scripts/bump_major_version.py

```diff
@@ -0,0 +1,335 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import argparse
+import shutil
+import yaml
+
+# With each new major release of CockroachDB, there are a number of
+# administrative tasks that need to be accomplished on the docs website.  This
+# includes, but is not limited to:
+
+# - Create a new `vXY.Z` folder, and copy content to it
+#
+# - Copy includes to a new `_includes/vXY.Z` folder
+#
+# - Copy images to a new versioned folder, and edit image link text in the
+# referencing documents as needed to point to the right places
+#
+# - Update the YAML config files with the new version info
+#
+# - Create a new sidebar JSON file
+#
+# - In the newly created XY.Z folder, remove the various "new in vXY.Z-1"
+# callouts from the copied files.
+#
+# Note: this list may not be exhaustive, see the code below for details.
+#
+# This script attempts to automate these tasks as much as possible.  In
+# addition to copying files around, editing files, etc., it also searches for
+# references to the previous version vX.Y.Z-1 in the newly copied files that
+# now serve as the documentation for vX.Y.Z, and shows those references to the
+# end user (as grep-like output), so they can be edited (or not) as needed.
+#
+# For example, to do all of the admin tasks around bumping the version of
+# CockroachDB from 20.1 to 20.2, and then surfacing the places where additional
+# edits may be needed, run:
+#
+# $ ./bump_major_version.py -d /path/to/docs -f 20.1 -t 20.2
+
+parser = argparse.ArgumentParser(
+    description="do admin tasks related to version bumps"
+)
+
+parser.add_argument("-f", "--from-version",
+                    required=True,
+                    help="previous version of CockroachDB")
+
+parser.add_argument("-t", "--to-version",
+                    required=True,
+                    help="next version of CockroachDB")
+
+parser.add_argument("-d", "--docs-directory",
+                    default=os.environ.get("COCKROACH_DOCS_REPO"),
+                    help="directory where you store 'cockroachdb/docs'")
+
+args = parser.parse_args()
+
+from_version = args.from_version
```

Comment:
Could you call `version_string_name` here from parsed args, so that `from_version` and `to_version` are strings instead of ints? That way you don't have to repeatedly call `version_string_name` later on within functions.

Reply:
Done.  There are now 2 effectively global vars, `old_version_name`, and `new_version_name`, that are created here and used everywhere instead.  They are not passed in as function arguments, so most functions take 0 or 1 args now, e.g., `copy_files(from_version, to_version)` becomes `copy_files()`, and so on.